### ISSUE TITLE
Returned exit code 12 and 'CommonPrefixes' for Empty S3 bucket.

### ISSUE THREAD

**Issue by RamiusKahn** [wazuh/wazuh#3715]
| Wazuh version | Component | Install type | Install method | Platform |
| --- | --- | --- | --- | --- |
| 3.9.3 | wazuh-modulesd:aws-s3 | Manager | Packages | Ubuntu18 |

Possibly related to issue: https://github.com/wazuh/wazuh/issues/2586

Wazuh Modulesd: AWS-S3 reports the following errors when configured to pull files from VPC flow log buckets in S3. An IAM role is configured for permissions to S3.

Configuration:
```
  <wodle name="aws-s3">
    <disabled>no</disabled>
    <remove_from_bucket>no</remove_from_bucket>
    <interval>10m</interval>
    <run_on_start>yes</run_on_start>
    <skip_on_error>yes</skip_on_error>
    <bucket type="cloudtrail">
      <name>cloudtrail</name>
    </bucket>
      <bucket type="vpcflow">
      <name>ge-dcos-cf-01-vpc-flow</name>
    </bucket>
    <bucket type="vpcflow">
      <name>misc-apps-vpc-flow</name>
    </bucket>
    <bucket type="vpcflow">
      <name>q-production-vpc-flow</name>
    </bucket>
    <bucket type="vpcflow">
      <name>sandbox-vpc-flow</name>
    </bucket>
    <bucket type="vpcflow">
      <name>elm-vpc-flow</name>
    </bucket>
  </wodle>
```

Error output:
```
2019/07/22 21:11:17 wazuh-modulesd:aws-s3: INFO: Module AWS started
2019/07/22 21:11:17 wazuh-modulesd:aws-s3: INFO: Starting fetching of logs.
2019/07/22 21:11:17 wazuh-modulesd:aws-s3: INFO: Executing Bucket Analysis: (Bucket: cloudtrail, Type: cloudtrail)
2019/07/22 21:11:18 wazuh-modulesd:aws-s3: INFO: Executing Bucket Analysis: (Bucket: ge-dcos-cf-01-vpc-flow, Type: vpcflow)
2019/07/22 21:11:19 wazuh-modulesd:aws-s3: INFO: Executing Bucket Analysis: (Bucket: misc-apps-vpc-flow, Type: vpcflow)
2019/07/22 21:11:20 wazuh-modulesd:aws-s3: INFO: Executing Bucket Analysis: (Bucket: q-production-vpc-flow, Type: vpcflow)
2019/07/22 21:11:21 wazuh-modulesd:aws-s3: INFO: Executing Bucket Analysis: (Bucket: sandbox-vpc-flow, Type: vpcflow)
2019/07/22 21:11:21 wazuh-modulesd:aws-s3: WARNING: Bucket:  -  Returned exit code 12
2019/07/22 21:11:21 wazuh-modulesd:aws-s3: WARNING: Bucket:  -  'CommonPrefixes'
2019/07/22 21:11:21 wazuh-modulesd:aws-s3: INFO: Executing Bucket Analysis: (Bucket: elm-vpc-flow, Type: vpcflow)
2019/07/22 21:11:22 wazuh-modulesd:aws-s3: INFO: Fetching logs finished.
```

Each of the other buckets are processing normally and CloudTrail and VPC flow logs are present in Kibana. 

Bucket `sandbox-vpc-flow` was automatically configured in the wodle config via script when the VPC was created in AWS, but has not been used yet and therefore has no logs, causing `WARNING: Bucket:  -  Returned exit code 12` and `WARNING: Bucket:  -  'CommonPrefixes'` to be thrown on each modulesd:aws-s3 run. It would be optimal to handle this condition without throwing an Error, possibly throwing something like `INFO: Bucket ($bucketName) Connected successfully, but contains no logs to process.` when the bucket exists, type is correct, is a valid bucket, permissions are correct, but no files exist. 

I think this is different than `bad bucket type` and `invalid bucket`.

**Comment by druizz90** [comment]
Hi @RamiusKahn,

We introduced a fix when the bucket type was wrong in #3652 for `3.10` version, but it will be improved after check your issue.

Thank you for your contribution.

Best regards,

Demetrio.

**Comment by crd1985** [comment]
Fixed in #3743 .
